extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSourceRuntimeConfig.java

```diff
@@ -130,4 +130,16 @@
      */
     @ConfigItem
     public Optional<String> validationQuerySql;
+
+    /**
+     * When enabled jdbc url will be forged to enable google socket factory
+     */
+    @ConfigItem(defaultValue = "false")
+    public boolean useGcp;
+
+    /**
+     * Query executed to validate a connection.
+     */
+    @ConfigItem
+    public Optional<String> cloudSqlInstance;
```

Comment:
isn't this replicating the `validationQuerySql` property ?

Reply:
Nope, this property value will end up in the jdbc url and will be used by the `SocketFactory` to provide access to the CloudSQL instance.